File: src/main/kotlin/br/com/caethas/atto/atto/modules/membro/service/MembroService.kt
package br.com.caethas.atto.atto.modules.membro.service

import br.com.caethas.atto.atto.modules.membro.dto.MembroDto
import br.com.caethas.atto.atto.modules.membro.entity.MembroEntity
import br.com.caethas.atto.atto.modules.membro.repository.MembroRepository
import br.com.caethas.atto.atto.shared.base.BaseService
import org.springframework.stereotype.Service
import java.util.*

@Service
class MembroService(
    repository: MembroRepository
) : BaseService<MembroEntity, MembroRepository>(repository) {

    /**
     * Busca membro pelo ID externo (membrosId) vindo do app
     */
    fun findByMembrosId(membrosId: UUID): MembroEntity =
        (repository as MembroRepository).findByMembrosId(membrosId)
            .orElseThrow { RuntimeException("Membro não encontrado com membrosId: $membrosId") }
}

File: lib/src/modules/membro/service/membro_service.dart
import 'package:atto_api/src/modules/membro/entity/membro_entity.dart';
import 'package:atto_api/src/modules/membro/repository/membro_repository_interface.dart';
import 'package:atto_api/src/modules/membro/service/membro_service_interface.dart';
import 'package:atto_api/src/shared/service/base_service.dart';
import 'package:vaden/vaden.dart';

@Service()
class MembroService extends BaseService<MembroEntity> implements IMembroService {
  final IMembroRepository repository;
  MembroService(this.repository) : super(repository);

  @override
  Future<MembroEntity> saveOrUpdate(MembroEntity entity) async {
    return await repository.saveOrUpdate(entity);
  }
  
}